api/cluster.go

```diff
@@ -174,3 +174,23 @@ func (cluster *Cluster) ChannelOverrides() (map[string]string, error) {
 	}
 	return result, nil
 }
+
+func (cluster *Cluster) KarpenterPools() []*NodePool {
+	var kp []*NodePool
+	for _, n := range cluster.NodePools {
+		if n.IsKarpenter() {
+			kp = append(kp, n)
+		}
+	}
+	return kp
+}
+
+func (cluster *Cluster) ClusterAutoscalerPools() []*NodePool {
+	var cp []*NodePool
+	for _, n := range cluster.NodePools {
+		if !n.IsKarpenter() {
+			cp = append(cp, n)
```

Comment:
Do we care that this would include master node pools which are not managed by CA? We could also rename `ClusterAutoscalerPools()` to something like `ASGBackedPools` which would be factually correct for masters and legacy workers.